test/EspressoSGXTEEVerifier.t.sol

```diff
@@ -0,0 +1,220 @@
+import {IEspressoSGXTEEVerifier} from "../src/interface/IEspressoSGXTEEVerifier.sol";
+
+contract EspressoSGXTEEVerifierTest is Test {
+    address proxyAdmin = address(140);
+    address adminTEE = address(141);
+    address fakeAddress = address(145);
+
+    EspressoSGXTEEVerifier espressoSGXTEEVerifier;
+    bytes32 reportDataHash =
+        bytes32(0x38f8abca50cdede6a00d405856857bc3d81135624ee0e287640956d11cc22d5e);
+    bytes32 enclaveHash =
+        bytes32(0x01f7290cb6bbaa427eca3daeb25eecccb87c4b61259b1ae2125182c4d77169c0);
+    bytes32 enclaveSigner =
+        bytes32(0x5fc862cb2e7e1f449f36a18b18aca08c20feaed0d411247816c281d596420cbb);
+    //  Address of the automata V3QuoteVerifier deployed on sepolia
+    address v3QuoteVerifier = address(0x6E64769A13617f528a2135692484B681Ee1a7169);
+
+    function setUp() public {
+        vm.createSelectFork("https://rpc.ankr.com/eth_sepolia");
+        // Get the instance of the DCAP Attestation QuoteVerifier on the Arbitrum Sepolia Rollup
+        vm.startPrank(adminTEE);
+        espressoSGXTEEVerifier =
+            new EspressoSGXTEEVerifier(enclaveHash, enclaveSigner, v3QuoteVerifier);
+        vm.stopPrank();
+    }
+
+    function testRegisterSigner() public {
+        vm.startPrank(adminTEE);
+        // bytes memory attestation = vm.readFileBinary("/test/configs/attestation.bin");
+        string memory quotePath = "/test/configs/attestation.bin";
+        string memory inputFile = string.concat(vm.projectRoot(), quotePath);
+        bytes memory sampleQuote = vm.readFileBinary(inputFile);
+
+        // take keccak256 hash of the address of proxyAdmin
+
+        address batchPosterAddress = address(0xe2148eE53c0755215Df69b2616E552154EdC584f);
+
+        bytes memory data = abi.encodePacked(batchPosterAddress);
+
+        // Convert the data to bytes32 and pass it to the verify function
+        espressoSGXTEEVerifier.registerSigner(sampleQuote, data);
+        vm.stopPrank();
+    }
+
+    function testRegisterSignerInvalidQuote() public {
+        vm.startPrank(adminTEE);
+        string memory quotePath = "/test/configs/invalid_quote.bin";
+        string memory inputFile = string.concat(vm.projectRoot(), quotePath);
+        bytes memory sampleQuote = vm.readFileBinary(inputFile);
+
+        address batchPosterAddress = address(0xe2148eE53c0755215Df69b2616E552154EdC584f);
+
+        bytes memory data = abi.encodePacked(batchPosterAddress);
+
+        vm.expectRevert(IEspressoSGXTEEVerifier.InvalidQuote.selector);
+        espressoSGXTEEVerifier.registerSigner(sampleQuote, data);
+    }
+
+    function testRegisterSignerInvalidAddress() public {
+        vm.startPrank(adminTEE);
+        string memory quotePath = "/test/configs/attestation.bin";
+        string memory inputFile = string.concat(vm.projectRoot(), quotePath);
+        bytes memory sampleQuote = vm.readFileBinary(inputFile);
+
+        address batchPosterAddress = address(0x4C91660a37d613E1Bd278F9Db882Cc5ED2549072);
```

Comment:
perhaps add another check with the zero addresss

Reply:
true, will add that!

Reply:
ok got it